common/quotas/global.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2022 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package quotas
+
+import (
+	"math"
+
+	"github.com/uber/cadence/common/membership"
+)
+
+// PerMember allows creating per instance RPS based on globalRPS averaged by member count for a given service.
+// If member count can not be retrieved or globalRPS is not provided it falls back to instanceRPS.
+func PerMember(service string, globalRPS, instanceRPS float64, resolver membership.Resolver) float64 {
+	if globalRPS > 0 {
+		memberCount, err := resolver.MemberCount(service)
+		if err == nil && memberCount > 0 {
+			avgQuota := math.Max(globalRPS/float64(memberCount), 1)
+			return math.Min(avgQuota, instanceRPS)
+		}
+	}
+	return instanceRPS
```

Comment:
suggestion to reduce nesting:
```suggestion
	if globalRPS < 1 {
		return instanceRPS
	}

	memberCount, err := resolver.MemberCount(service)
	if err != nil || memberCount < 1 {
		return instanceRPS
	}

	avgQuota := math.Max(globalRPS/float64(memberCount), 1)
	return math.Min(avgQuota, instanceRPS)
```